packages/next-app/src/app/metadata/(components)/SearchPoolForm.tsx

```diff
@@ -0,0 +1,121 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import React from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useAccount, useNetwork } from "wagmi";
+
+import { Button } from "#/components/Button";
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import balancerGql, { networkIdEnumMap } from "#/lib/gql";
+
+export interface PoolMetadataAttribute {
+  poolId: string;
+  network: string;
+}
+
+const inputTypenames = [
+  { value: "1", label: "Mainnet" },
+  { value: "137", label: "Polygon" },
+  { value: "42161", label: "Arbitrum" },
+  { value: "5", label: "Goerli" },
+];
+
+export default function SearchPoolForm({ close }: { close?: () => void }) {
+  const {
+    register,
+    handleSubmit,
+    setError,
+    control,
+    watch,
+    clearErrors,
+    resetField,
+    formState: { errors },
+  } = useForm<PoolMetadataAttribute>();
+  const { push } = useRouter();
+  const { chain } = useNetwork();
+  const { isConnected } = useAccount();
+
+  const poolId = watch("poolId");
+  const network = isConnected ? chain?.id.toString() : watch("network");
+
+  const { data: poolAddress } = balancerGql(network || "1").usePoolAddress(
+    {
+      poolId,
+    },
+    { revalidateIfStale: true }
+  );
+
+  const isPool = poolAddress?.pool?.address ? true : false;
+
+  function handleSubmitForm(formData: PoolMetadataAttribute) {
+    const networkId = formData.network ?? chain?.id.toString();
+    const networkName =
+      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
+    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+
+    close?.();
+  }
+
+  React.useEffect(() => {
+    if (poolAddress && !poolAddress.pool && poolId) {
+      setError(
+        "poolId",
+        {
+          type: "notfound",
+          message: "Pool not found. Check the Pool ID and network.",
+        },
+        { shouldFocus: true }
+      );
+      return;
+    } else {
+      clearErrors("poolId");
+    }
```

Comment:
Any reason to set error here instead of doing it on the form submit?

Reply:
The button will be disabled until the user enters a valid pool id, which must correspond to an existing pool

Reply:
This is a bit unintuitive to me 😕  you're searching a pool but you can't search unless it already exists?

Reply:
It was a requirement we planned for the task. 

`"go to pool disabled until validation that the pool exists runs;"`

Reply:
Right -- this should probably be called `Go to pool` instead of search then, WDYT? Ultimately we're not showing the user a list of results, but directing them to a specific pool page

Reply:
Agreed